tests/prefetch_related/tests.py

```diff
@@ -291,6 +292,15 @@ def test_m2m_then_reverse_one_to_one_object_ids(self):
         sql = queries[-1]['sql']
         self.assertWhereContains(sql, self.author1.id)
 
+    def test_filter_deferred(self):
+        add_q = Query.add_q
```

Comment:
I think these tests could the rare cases where a docstring explaining what it tests could be valuable given the large amount of boilerplate required.

What about

```
"""Related filtering of prefetched querysets is deferred until necessary."""
```

Reply:
Sounds good, will add